Source/Scene/ImageryLayer.js

```diff
@@ -866,6 +971,81 @@ define([
         imagery.state = ImageryState.TEXTURE_LOADED;
     };
 
+    /**
+     * Creates WebGL textures for each image in a given multi-source {@link Imagery} instance.
+     * @private
+     *
+     * @param {Context} context The rendered context to use to create textures.
+     * @param {Imagery} imagery The imagery for which to create a texture.
+     * @param
+     */
+    ImageryLayer.prototype._createMultipleTextures = function(context, imagery) {
+        var imageryProvider = this._imageryProvider;
+        var projectedImages = imagery.projectedImages;
+
+        // If this imagery provider has a discard policy, use it to check if this
+        // image should be discarded.
+        if (defined(imageryProvider.tileDiscardPolicy)) {
+            var discardPolicy = imageryProvider.tileDiscardPolicy;
+            if (defined(discardPolicy)) {
+                // If the discard policy is not ready yet, transition back to the
+                // RECEIVED state and we'll try again next time.
+                if (!discardPolicy.isReady()) {
+                    imagery.state = ImageryState.RECEIVED;
+                    return;
+                }
+
+                // Mark discarded imagery tiles invalid.  Parent imagery will be used instead.
+                if (discardPolicy.shouldDiscardImage(projectedImages[0])) {
+                    imagery.state = ImageryState.INVALID;
+                    return;
+                }
+            }
+        }
+
+        //>>includeStart('debug', pragmas.debug);
+        if (this.minificationFilter !== TextureMinificationFilter.NEAREST &&
+            this.minificationFilter !== TextureMinificationFilter.LINEAR) {
+            throw new DeveloperError('ImageryLayer minification filter must be NEAREST or LINEAR');
+        }
+        //>>includeEnd('debug');
+
+        var sampler = new Sampler({
+            minificationFilter : this.minificationFilter,
+            magnificationFilter : this.magnificationFilter
+        });
+
+        var projectedImagesLength = projectedImages.length;
+
+        for (var i = 0; i < projectedImagesLength; i++) {
+            var image = projectedImages[i];
+
+            var texture;
+            if (defined(image.internalFormat)) {
```

Comment:
Do you expect this to load compressed textures? Wouldn't all of the images be `HTMLImageElement`s?

Reply:
This is what `ImageryLayer.prototype._createTexture` does, although it looks like that should just be loading `HTMLImageElement`s too. Maybe it was originally intended for greyscale imagery somehow?